### ISSUE TITLE
Postcode recognition in France

### ISSUE THREAD

**Issue by lamquiem** [openvenues/libpostal#638]
Hi!

I was checking out libpostal, and saw something that could be improved.

---
#### My country is France
---
#### Here's how I'm using libpostal
We use libpostal to parse addresses before searching with elasticsearch.
---
#### Here's what I did
parse_address('1 rue saint roch 2B238 poggio-di-venaco',language = 'fr', country = 'fr')

---
#### Here's what I got
[('1', 'house_number'),
 ('rue saint roch 2b238', 'road'),
 ('poggio-di-venaco', 'city')]

---
#### Here's what I was expecting
[('1', 'house_number'),
 ('rue saint roch', 'road'),
('2b238','postcode'),
 ('poggio-di-venaco', 'city')]

---
#### For parsing issues, please answer "yes" or "no" to all that apply.

- Does the input address exist in [OpenStreetMap](https://openstreetmap.org)?
  no
- Do all the toponyms exist in OSM (city, state, region names, etc.)?
  yes
- If the address uses a rare/uncommon format, does changing the order of the fields yield the correct result?
  no
---
#### Here's what I think could be improved
Is it possible to specify that French postcodes are of the form (\\d[0-9aAbB]\d{3}) when parsing? 
The codes '2A' and '2B' correspond to the two Corsican departments in France. Openstreet map treats them as '20' but this is not the reality.
Is it possible to set libpostal to recognise this form of regex ?

**Comment by prigaux** [comment]
https://fr.wikipedia.org/wiki/Poggio-di-Venaco says postcode is

**Comment by albarrentine** [comment]
yes guessing that postcode format doesn't exist in the training data (you can type `.print_features` in the address_parser cli to see what the model is doing and what might work). 20250 works

1 rue saint roch 20250 poggio-di-venaco
```
{
  "house_number": "1",
  "road": "rue saint roch",
  "postcode": "20250",
  "city": "poggio-di-venaco"
}
```

You can use a regex to extract/remove postcodes following that pattern and reparse the remainder, e.g. something like this will usually also work. If you're sending to Elasticsearch, you can just add the extracted postcode back in at the end if it improves the results.

1 rue saint roch poggio-di-venaco
```
{
  "house_number": "1",
  "road": "rue saint roch",
  "city": "poggio-di-venaco"
}
```
